File: tests/coretests/Stocks/Services/VolumeAnalysisTests.cs
using System.Collections.Generic;
using System.Linq;
using core.fs.Services;
using core.fs.Services.Analysis;
using coretests.testdata;
using Xunit;

namespace coretests.Stocks.Services;

public class VolumeAnalysisTests
{
    private readonly IEnumerable<AnalysisOutcome> _outcomes = MultipleBarPriceAnalysis.VolumeAnalysis.generate(
        TestDataGenerator.IncreasingPriceBars(numOfBars: 300));
    
    [Fact]
    public void VolumeAnalysis_Adds_AllOutcomes() =>
        Assert.Single(_outcomes);

    private void OutcomeExistsAndValueMatches(string key, decimal value)
    {
        Assert.Contains(_outcomes, o => o.Key == key);
        Assert.Equal(value, _outcomes.First(o => o.Key == key).Value);
    }
    
    [Fact]
    public void AverageVolume_PresentAndValid() =>
        OutcomeExistsAndValueMatches(MultipleBarPriceAnalysis.MultipleBarOutcomeKeys.AverageVolume, 269);
}

File: tests/coretests.fs/Stocks/Services/VolumeAnalysisTests.fs
module coretests.fs.Stocks.Services.VolumeAnalysisTests

open Xunit
open core.fs.Services
open FsUnit
open core.fs.Services.Analysis
open testutils

let outcomes = TestDataGenerator.IncreasingPriceBars(300)
                |> MultipleBarPriceAnalysis.VolumeAnalysis.generate

[<Fact>]
let ``VolumeAnalysis Adds AllOutcomes`` () =
    outcomes |> Seq.length |> should equal 1
    
[<Fact>]
let ``AverageVolume PresentAndValid`` () =
    outcomes |> Seq.head |> _.Value |> should equal 269m
    outcomes |> Seq.head |> _.Key |> should equal MultipleBarPriceAnalysis.MultipleBarOutcomeKeys.AverageVolume
